cmd/run.go

```diff
@@ -70,17 +73,32 @@ func run() func(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("failed to parse config: %w", err)
 		}
 
-		log := logger.NewLogger()
-		ctx := logger.IntoContext(context.Background(), log)
+		ctx, cancel := logger.NewContextWithLogger(context.Background())
+		log := logger.FromContext(ctx)
+		defer cancel()
 
 		if err = cfg.Validate(ctx); err != nil {
 			return fmt.Errorf("error while validating the config: %w", err)
 		}
 
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
 		s := sparrow.New(cfg)
+		cErr := make(chan error, 1)
 		log.Info("Running sparrow")
-		if err = s.Run(ctx); err != nil {
-			err = fmt.Errorf("error while running sparrow: %w", err)
+		go func() {
+			if err := s.Run(ctx); err != nil {
+				cErr <- fmt.Errorf("error while running sparrow: %w", err)
+			}
+		}()
+
+		select {
+		case <-sigChan:
+			log.Info("Signal received, shutting down")
+			cancel()
+			<-cErr
+		case err := <-cErr:
 			return err
 		}
```

Comment:
What do you think about this? First I've only implemented it here for testing purposes of a context cancellation but then I thought, why not handle OS signals? 😄

Reply:
I've opened the PR for this: #126